src/MLJInterface.jl

```diff
@@ -0,0 +1,211 @@
+        runtests::Bool = true
+        loss_type::Type = Nothing
+        selection_method::Function = choose_best
+    end)
+# TODO: To reduce code re-use, we could forward these defaults from
+#       `equation_search`, similar to what we do for `Options`.
+
+"""Generate an `SRRegressor` struct containing all the fields in `Options`."""
+function modelexpr()
+    struct_def = deepcopy(sr_regressor_template)
+    fields = last(last(struct_def.args).args).args
+
+    # Add everything from `Options` constructor directly to struct:
+    for (i, option) in enumerate(DEFAULT_OPTIONS)
+        insert!(fields, i, Expr(:(=), option.args...))
+    end
+
+    # We also need to create the `get_options` function, based on this:
+    constructor = :(Options(;))
+    constructor_fields = last(constructor.args).args
+    for option in DEFAULT_OPTIONS
+        symb = getsymb(first(option.args))
+        push!(constructor_fields, Expr(:kw, symb, Expr(:(.), :m, Core.QuoteNode(symb))))
+    end
+
+    return quote
+        $struct_def
+        function get_options(m::SRRegressor)
+            return $constructor
+        end
+    end
+end
+function getsymb(ex::Symbol)
+    return ex
+end
+function getsymb(ex::Expr)
+    for arg in ex.args
+        isa(arg, Symbol) && return arg
+        s = getsymb(arg)
+        isa(s, Symbol) && return s
+    end
+    return nothing
+end
+
+"""Get an equivalent `Options()` object for a particular regressor."""
+function get_options(::AbstractSRRegressor) end
+
+eval(modelexpr())
+
+# Cleaning already taken care of by `Options` and `equation_search`
+function full_report(m::AbstractSRRegressor, fitresult)
+    _, hof = fitresult.state
+    # TODO: Adjust baseline loss
+    formatted = format_hall_of_fame(hof, fitresult.options, 1.0)
+    equation_strings = get_equation_strings(formatted.trees, fitresult.options)
+    best_idx = dispatch_selection(
+        m.selection_method,
+        formatted.trees,
+        formatted.losses,
+        formatted.scores,
+        formatted.complexities,
+    )
+    return (;
+        best_idx=best_idx,
+        equations=formatted.trees,
+        equation_strings=equation_strings,
+        losses=formatted.losses,
+        complexities=formatted.complexities,
+        scores=formatted.scores,
+    )
+end
+
+# TODO: Pass `variable_names` and `units`
+# TODO: Enable `verbosity` being passed to `equation_search`
+MMI.clean!(::AbstractSRRegressor) = ""
+
+function MMI.fit(m::AbstractSRRegressor, verbosity, X, y, w=nothing)
+    return _update(m, verbosity, (; state=nothing), nothing, X, y, w)
+end
+function MMI.update(
+    m::AbstractSRRegressor, verbosity, old_fitresult, old_cache, X, y, w=nothing
+)
+    return _update(m, verbosity, old_fitresult, old_cache, X, y, w)
+end
+function _update(m::AbstractSRRegressor, verbosity, old_fitresult, old_cache, X, y, w)
+    options = get(old_fitresult, :options, get_options(m))
+    X_t = transpose(MMI.matrix(X))
```

Comment:
Note to self: should ensure that this makes a copy rather than a view. Would want to avoid accessing feature slices of data in row-major format as it would be slower.